runtime/doc/api.txt

```diff
@@ -3196,6 +3196,15 @@ nvim_open_win({buffer}, {enter}, {config})                   *nvim_open_win()*
                   • focusable: Enable focus by user actions (wincmds, mouse
                     events). Defaults to true. Non-focusable windows can be
                     entered by |nvim_set_current_win()|.
+                  • mouse: Specify how this window affects mouse events. May
+                    be a boolean or a Lua callback. Defaults to `focusable`
+                    value.
```

Comment:
It may be hard to describe the default value here:
- When `mouse` was `true` or `false`, and `focusable` is specified in `nvim_win_set_config` while `mouse` is not, `mouse` needs to be set to the same value as `focusable` for backward-compatibility.
- However, when `mouse` was a callback, it's better not to change it when `focusable` is specified in `nvim_win_set_config`.

Reply:
I was originally thinking that it would be good to track if `mouse` had ever been manually set, and if it had been then don't automatically change it to follow `focusable`. But doing as you describe is nice and simple.

It means that if you ever manually set `mouse` to something different than `focusable`, it is probably best to always set it when setting `focusable`. That's reasonable and simple.

I guess it's friendly to not automatically change `mouse` when it's a callback; but it doesn't seem to offer that much and makes things more difficult to describe. Instead of special casing callback, in `nvim_win_set_config` could just say something like:
"If `focusable` is specified and `mouse` is not, then `mouse` is set to the same value as `focusable`."